### PR TITLE
Documented everything in UserForm and it's sub-components with commas…

### PR DESCRIPTION
**Pull request by isaaclee12**
…, wrote pseudocode, made a todo list, and an issues list.

# Description

Main issue is yet to be fixed, this is just a version of the components we'll be working on with descriptive comments

## Type of change

Please select everything applicable. Please, do not delete any lines.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Issue
- [ ] Is this related to a specific issue? Is so please specify:

# Checklist:

- [ X ] I have executed `npm run lint` and resolved any outstanding errors. Most issues can be solved by executing `npm run format`
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ X ] I have commented my code, particularly in hard-to-understand areas
- [ X ] My changes generate no new warnings

### CODE REVIEW

**Review by Caleb-Cohen — CHANGES_REQUESTED** [review]

**Comment by Caleb-Cohen** [comment]
@WonjaeSung and @DevRhyse - I'm not sure what type of work you've done on the user form, but this will probably generate a few conflicts if either of y'all have pushed work to your local. Can I get a sense of where y'all are at before merging this?

**Comment by WonjaeSung** [comment]
Hey @Caleb-Cohen,

@isaaclee12, @DevRhyse and I have worked on this yesterday and was able to fix below issues from #153 :

-Start dates cannot be in the past

-Start or end dates can only exist for 2022/2023. No 2024+ or before 2022

With that said, I have pushed our progress to my local. What's the best way to merge this? The only file we worked on was FormCreateEvent.js.

**Comment by Caleb-Cohen** [comment]
> Hey @Caleb-Cohen,
> 
> @isaaclee12, @DevRhyse and I have worked on this yesterday and was able to fix below issues from #153 :
> 
> -Start dates cannot be in the past
> 
> -Start or end dates can only exist for 2022/2023. No 2024+ or before 2022
> 
> With that said, I have pushed our progress to my local. What's the best way to merge this? The only file we worked on was FormCreateEvent.js.

You'll do a PR to the same branch. There will be conflicts with this, but we have two options. 

1. I merge this PR. You submit a PR, I resolve the conflicts, and then approve the merge. 
2. I merge this PR. You pull the update into your local and resolve the conflicts. Commit the changes, then push and PR back to this branch.

Decision criteria come down to who you want to handle the conflicts.

File: client/src/features/form/UserForm.js [hunk 1]
```diff
@@ -1,40 +1,119 @@
...
   const [userData, setUserData] = useState({
     recurring: { rate: "noRecurr", days: [] },
   });
+
+  /* Notes about how the form works:
+  Start Date is the default date for a one-off singular event
+  End Date is a date you can set IF you are recurring.
+  */
+
+  // ISAAC'S ISSUE TO SOLVE:
```
Comment (Caleb-Cohen) [inline client/src/features/form/UserForm.js:34]: @isaaclee12 I'm ok with all this for now, but the finished PR into development will need to have this cleaned up.

Comment (isaaclee12) [reply client/src/features/form/UserForm.js:34]: Noted

**Comment by isaaclee12** [comment]
@Caleb-Cohen We just discussed this, we're going to go with option 2: You merge this PR, then pull the update and resolve the conflicts. Then we'll commit the changes, then push and PR back to this branch. Much thanks

**Comment by Caleb-Cohen** [comment]
> @Caleb-Cohen We just discussed how to deal with merging isaaclee12:feature/softSecurity-commented into Caleb-Cohen:feature/softSecurity.
> 
> We're going to go with option 2: You merge this PR, then pull the update and resolve the conflicts. Then we'll commit the changes, then push and PR back to this branch. Much thanks :)

will do! It's a bit of a learning experience, but a good one! feel free to reach out if you run into any problems.

**Pull request merged**